packages/fxa-shared/payments/stripe.ts

```diff
@@ -466,44 +469,74 @@ export abstract class StripeHelper {
       }
     }
 
-    if (this.contentfulManager) {
-      try {
-        const contentfulToStripeMapper = new StripeMapperService(
-          this.contentfulManager
+    if (this.config.contentful!.enabled) {
```

Comment:
Unfortunately, I'm not sure using the feature flag here will have the desired result. 

With this implementation, even when the feature flag is enabled, the `StripeMapperService` will still use the Stripe Metadata as fallback when Contentful values aren't available. (This isn't desired, since with the feature flag set to true, only Contentful should be used, without Stripe fallback)

With that in mind, I'd recommend moving the feature flag into the StripeMapperService + PlanMapperUtil, and updating the logic there? (`PlanMapperUtil.mapField` seems like the best candidate)

Reply:
Updated as per our Slack chat. Thank you!